src/main/java/com/linecorp/armeria/server/thrift/ThriftServiceCodec.java

```diff
@@ -559,4 +559,13 @@ protected TProtocol initialValue() {
             return protoFactory.getProtocol(new TByteBufTransport());
         }
     }
+
+    private static Set<Class<?>> getAllInterfaces(Class<?> clz) {
+        Set<Class<?>> interfaces = new HashSet<>();
+        Collections.addAll(interfaces, clz.getInterfaces());
+        if (clz.getSuperclass() != null) {
+            interfaces.addAll(getAllInterfaces(clz.getSuperclass()));
```

Comment:
Yeah wasn't worrying too much about performance since this is all heavily reflective and only happens once, but guess it's not so hard to use the shared set. Anyways I realized I forgot about superinterfaces, so just went with the apache implementation.